CDCgov/prime-reportstream
# 14413: fetch all conditions for mapping at once

This PR updates the condition mapping to use a table saw filter as well as doing all the mappable codes for an observation in one pass

Test Steps:
1. Verify all the tests pass
2. Run a ~600 line bulk FHIR through the pipeline and note the timing of "process"
3. Run the same thing on this branch and note the timing
4. Confirm that it runs faster on this branch

## Changes
- adds an `isIn` operator to LookupTable FilterBuilder
- changes the condition mapping code to use table saw filters directly

Duration improvement
> After finally getting to running some durations I found that the process function goes from taking ~290 milliseconds to ~90 milliseconds

## Checklist

### Testing

### Process

## Linked Issues
- #14413 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
